src/coprocessor/dag/rpn_expr/impl_control.rs

```diff
@@ -14,8 +14,30 @@ fn if_null<T: Evaluable>(lhs: &Option<T>, rhs: &Option<T>) -> Result<Option<T>>
     Ok(rhs.clone())
 }
 
+#[rpn_fn(raw_varg)]
+#[inline]
+pub fn case_when<T: Evaluable>(args: &[ScalarValueRef<'_>]) -> Result<Option<T>> {
+    for chunk in args.chunks(2) {
+        if chunk.len() == 1 {
+            // else statement
+            // TODO: Must verify type
```

Comment:
With this concern, can we land this PR now? An invalid request just makes it panic.

Reply:
Invalid requests make other RPN function (including fixed args ones) panic already 😅 For example, pass in an `Int` while the function accept `Real`.

Reply:
Invalid requests make other RPN function (including fixed args ones) panic already 😅 For example, pass in an `Int` while the function accept `Real`.

Reply:
A fully functional validation will be resolved by https://github.com/tikv/tikv/pull/5033